apps/labs/posts/libsf-error-state.md

```diff
@@ -0,0 +1,744 @@
+  install: true
+)
+```
+
+and adding a `link_with` entry to the creation of each extension module:
+
+```
+py3.extension_module('_special_ufuncs',
+  ['_special_ufuncs.cpp', '_special_ufuncs_docs.cpp', 'sf_error.cc'],
+  include_directories: ['../_lib', '../_build_utils/src'],
+  dependencies: [np_dep],
+  link_with: [sf_error_state_lib], # The new line
+  link_args: version_link_args,
+  install: true,
+  cpp_args: ['-DSP_SPECFUN_ERROR'],
+  subdir: 'scipy/special'
+)
+```
+
+However, on my Linux machine SciPy would build without issue, but attempting to import `scipy.special` would
+result in the error:
+
+```
+ImportError: libsf_error_state.so: cannot open shared object file: No such file or directory
+```
+
+After a period of head scratching in which I pondered every possible explanation except the correct one, I showed Irwin
+what I tried and the error message I was getting. It turned out that the difference in operating systems was a red
+herring. The issue was that of the two methods for building SciPy from source for development recommended in SciPy's
+[contributor
+documentation](https://docs.scipy.org/doc/scipy/building/index.html#building-from-source-for-scipy-development), I was
+using the `"python dev.py build"` workflow and he was using an editable install: `"pip install -e . --no-build-isolation"`.
+For the editable install, SciPy is installed directly in its own project folder, and the shared library and the relevant
+extension modules were all being installed next to each other in `~/scipy/scipy/special`. For the `dev.py` workflow,
+SciPy is installed elsewhere. Since I didn't specify where to install the shared library, it got put in the wrong
+place. It turned out what I had to do was configure the `install_dir` in `meson` like this:
+
+```
+sf_error_state_lib = shared_library('sf_error_state', # Name of the library
+  # Implementation files contained it.
+  ['sf_error_state.c'],
+  # Additional places the preprocessor can look for header files
+  include_directories: ['../_lib', '../_build_utils/src'],
+  install: true
+  # Tell meson where to install the shared library
+  install_dir: py3.get_install_dir() / 'scipy/special',
+)
+```
+
+But even after this, I was still receiving the same error. After consulting `meson`'s excellent documentation and
+looking at some related issues, it turns out that the `pip` takes care to set the
+`RPATH` [] for each extension module, which tells the dynamic linker where to look
+for shared libraries.
+
+To get things to work with `dev.py`, I needed to explicitly set the `RPATH` in `meson` by adding
+`install_rpath: '$ORIGIN'` to each extension module. `'$ORIGIN'` means to search in the same folder as the extension
+module.
+
+### Building Wheels on Windows
+
+After setting `install_dir` and `install_rpath` correctly, all but one of SciPy's CI jobs were passing. The sole failing
+job involved building a wheel on Windows. A
+wheel can be thought of as a
```

Comment:
nit: could reflow lines here

Reply:
I had my editor set to wrap everything at 120 lines. I can change it to a more conventional 80 after everything settles, to avoid messing up the diff before then.